src/least_satisfying.rs

```diff
@@ -105,111 +93,111 @@ mod tests {
             }
         }
 
-        let res = least_satisfying(&satisfies_v, true, |i| *i);
+        let res = least_satisfying(&satisfies_v, |i| *i);
         let exp = first_yes.unwrap();
         TestResult::from_bool(res == exp)
     }
 
     #[test]
     fn least_satisfying_1() {
         assert_eq!(
-            least_satisfying(&[No, Unknown, Unknown, No, Yes], true,|i| *i),
+            least_satisfying(&[No, Unknown, Unknown, No, Yes],|i| *i),
             4
         );
     }
 
     #[test]
     fn least_satisfying_1f() {
```

Comment:
these  variant tests you added in the previous commit, which as far as I can tell were to ensure that the introduction of `start_check` did not misbehave, now serve no purpose. IMO you can eliminate them, either from this commit, or from the earlier commit that introduced `start_check`.

Reply:
They will go away as you work your way through the commits... :)

Reply:
nope, they are in the diff. :)

Reply:
Hmmm... OK let me check.  Apologies.

Reply:
(maybe you have another local commit that you still need to push up to github?)

Reply:
I am showing that I removed it in 8227356a2bf0640fc1eeb427dea568d544ead77d

Reply:
Sorry, sorry.  I misunderstood this issue.  Yes, I do need to remove the duplicated tests.  I thought that you were pointing to the boolean that was added to the tests.  I forgot that I duplicated all of them.  I will definitely remove them.

Reply:
d1e7eb8fce8afb9f713a85b372462e4aab6f8082